brickstrap.sh

```diff
@@ -19,29 +20,51 @@
 # http://twitter.com/kvz
 #
 
-### Configuration
+### Runtime
 #####################################################################
+set -e
+
+# Bash will remember & return the highest exitcode in a chain of pipes.
+set -o pipefail
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ] && LOG_LEVEL="3" # 4 = debug -> 0 = fail
 
-# Commandline options. This defines the usage page
+#
+# Get the brickstrap source directory
+#
+function br_script_path()
+{
+    SCRIPT_PATH=$(readlink -f "$0")
+    SCRIPT_PATH=$(dirname "$SCRIPT_PATH")
+    echo -n "$SCRIPT_PATH"
+}
 
-read -r -d '' usage <<-'EOF'
-  usage: brickstrap [-b <board>] [-d <dir>] <command>
+. "$(br_script_path)/brickstrap-components.sh"
+
+# Commandline options. This defines the usage page
+BRP_USAGE=$(cat <<'EOF'
+Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
```

Comment:
Sanity checks are as follows...

* Creation commands:
    * Checks for valid project and component source directories
    * Checks that `<dir>` does not exist
       * If no `-f` option, fails with error.
       * if yes `-f` option, runs `delete` command before proceeding
* Incremental build
    * Checks that `<dir>` is sane, e.g. has `brickstrap.conf`
    * Checks that project and components in brickstrap.conf are still valid
    * Checks that output of command does not already exist.
       * if no `-f` option, fail with error
      * if yes `-f` option, delete previous output and then proceed
* Debug:
    * check that `<dir>` is sane. e.g. has multistrap.conf for simulate-multistrap or has rootfs for run-hook/shell
* Delete:
    * check that `<dir>` exists